src/android/com/microsoftopentechnologies/intellij/helpers/azure/sdk/AzureSDKManagerImpl.java

```diff
@@ -52,40 +77,79 @@ public static AzureSDKManager getManager() {
             apiManagerADAuth = new AzureSDKManagerADAuthDecorator(apiManager);
         }
 
-        AzureAuthenticationMode authenticationMode = AzureRestAPIManager.getManager().getAuthenticationMode();
-        if(authenticationMode == AzureAuthenticationMode.ActiveDirectory) {
+        if (AzureRestAPIManager.getManager().getAuthenticationMode() == AzureAuthenticationMode.ActiveDirectory) {
             return apiManagerADAuth;
+        } else {
+            return apiManager;
         }
-
-        return apiManager;
     }
 
     @NotNull
     @Override
-    public List<VirtualMachine> getVirtualMachines(@NotNull String subscriptionId) throws AzureCmdException {
-        List<VirtualMachine> vmList = new ArrayList<VirtualMachine>();
+    public List<CloudService> getCloudServices(@NotNull String subscriptionId) throws AzureCmdException {
+        List<CloudService> csList = new ArrayList<CloudService>();
         ComputeManagementClient client = null;
 
         try {
-            client = AzureSDKHelper.getComputeManagementClient(subscriptionId);
+            client = getComputeManagementClient(subscriptionId);
 
-            if (client == null) {
-                throw new Exception("Unable to instantiate Compute Management client");
+            ArrayList<HostedService> hostedServices = getHostedServices(client).getHostedServices();
+
+            if (hostedServices == null) {
+                return csList;
             }
 
-            HostedServiceOperations hso = client.getHostedServicesOperations();
+            for (HostedService hostedService : hostedServices) {
+                String productionDeployment = getDeployment(
+                        client,
+                        hostedService.getServiceName(),
+                        DeploymentSlot.Production).getName();
+
+                String stagingDeployment = getDeployment(
+                        client,
+                        hostedService.getServiceName(),
+                        DeploymentSlot.Staging).getName();
+
```

Comment:
Can we run the 2 calls to `getDeployment` in parallel somehow? Both are I/O calls and can be run simultaneously.